plugins/scaffolder/src/alpha/extensions.tsx

```diff
@@ -26,31 +26,83 @@ import {
   identityApiRef,
   NavItemBlueprint,
   PageBlueprint,
+  swappableComponentsApiRef,
 } from '@backstage/frontend-plugin-api';
 import { rootRouteRef } from '../routes';
 import CreateComponentIcon from '@material-ui/icons/AddCircleOutline';
 import { FormFieldBlueprint } from '@backstage/plugin-scaffolder-react/alpha';
 import { scmIntegrationsApiRef } from '@backstage/integration-react';
 import { scaffolderApiRef } from '@backstage/plugin-scaffolder-react';
 import { ScaffolderClient } from '../api';
+import {
+  SwappableReviewStepComponent,
+  SwappableTaskPageComponent,
+  SwappableTemplateCardComponent,
+  SwappableTemplateListPageComponent,
+  SwappableTemplateOutputsComponent,
+  SwappableTemplateWizardPageComponent,
+} from './swappableComponents';
 
 export const scaffolderPage = PageBlueprint.makeWithOverrides({
   inputs: {
     formFields: createExtensionInput([
       FormFieldBlueprint.dataRefs.formFieldLoader,
     ]),
   },
-  factory(originalFactory, { inputs }) {
+  config: {
+    schema: {
+      headerOptions: z =>
+        z
+          .object({
+            pageTitleOverride: z.string().optional(),
+            title: z.string().optional(),
+            subtitle: z.string().optional(),
+          })
+          .optional(),
+      defaultPreviewTemplate: z => z.string().optional(),
+      contextMenu: z =>
+        z
+          .object({
+            editor: z.boolean().optional(),
+            actions: z.boolean().optional(),
+            tasks: z.boolean().optional(),
+            create: z.boolean().optional(),
+            templatingExtensions: z.boolean().optional(),
+          })
+          .optional(),
+    },
+  },
+  factory(originalFactory, { inputs, config, apis }) {
     const formFieldLoaders = inputs.formFields.map(i =>
       i.get(FormFieldBlueprint.dataRefs.formFieldLoader),
     );
+    const api = apis.get(swappableComponentsApiRef);
+    const reviewStepComponentExists = api?.componentIsLoadable(
+      SwappableReviewStepComponent.ref,
+    );
     return originalFactory({
       routeRef: convertLegacyRouteRef(rootRouteRef),
       path: '/create',
       loader: () =>
         import('../components/Router/Router').then(m =>
           compatWrapper(
-            <m.InternalRouter formFieldLoaders={formFieldLoaders} />,
+            <m.InternalRouter
+              {...config}
+              formFieldLoaders={formFieldLoaders}
+              components={{
+                TemplateCardComponent: SwappableTemplateCardComponent,
+                ReviewStepComponent: reviewStepComponentExists
```

Comment:
I'm not sure that this is strictly necessary.

We should just have the default implementation baked into the `swappableComponent` and just always pass that through. That way it will render the swapped one, or the default one.